### ISSUE TITLE
Replace grep for 'import google' with checkstyle plugin

### ISSUE THREAD

**Issue by spencergibb** [spring-cloud/spring-cloud-build#18]
Tweet that spawned the issue: https://twitter.com/phillip_webb/status/634106050312953857 thanks @philwebb 

The grep check. https://github.com/spring-cloud/spring-cloud-build/blob/master/spring-cloud-build/pom.xml#L284-L317

**Comment by philwebb** [comment]
Here's one I just added to a project to ensure BDD mockito is used. Same principal would apply for Guava.

`pom.xml`

```
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-checkstyle-plugin</artifactId>
				<version>2.16</version>
				<executions>
					<execution>
						<id>validate</id>
						<phase>validate</phase>
						<configuration>
							<configLocation>checkstyle.xml</configLocation>
							<encoding>UTF-8</encoding>
							<consoleOutput>true</consoleOutput>
							<failsOnError>true</failsOnError>
							<includeTestSourceDirectory>true</includeTestSourceDirectory>
						</configuration>
						<goals>
							<goal>check</goal>
						</goals>
					</execution>
				</executions>
			</plugin>
```

`checkstyle.xml`
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.2//EN" "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">
<module name="Checker">
	<module name="TreeWalker">
		<module name="AvoidEscapedUnicodeCharacters">
			<property name="allowEscapesForControlCharacters" value="true" />
			<property name="allowByTailComment" value="true" />
			<property name="allowNonPrintableEscapes" value="true" />
		</module>
		<module name="ImportControl">
			<property name="file" value="import-control.xml" />
		</module>
		<module name="RedundantImport" />
		<module name="UnusedImports">
			<property name="processJavadoc" value="true" />
		</module>
		<module name="AvoidStarImport" />
		<module name="RequireThis">
			<property name="checkMethods" value="false" />
		</module>
		<module name="StringLiteralEquality" />
		<module name="MultipleVariableDeclarations" />
		<module name="OneStatementPerLine" />
		<module name="NeedBraces" />
		<module name="GenericWhitespace" />
		<module name="OneTopLevelClass" />
		<module name="MissingSwitchDefault" />
		<module name="NoFinalizer" />
		<module name="GenericWhitespace" />
	</module>
	<module name="FileTabCharacter" />
</module>
```

`import-control.xml`

```
<?xml version="1.0"?>
<!DOCTYPE import-control PUBLIC "-//Puppy Crawl//DTD Import Control 1.1//EN" "http://www.puppycrawl.com/dtds/import_control_1_1.dtd">
<import-control pkg="com.aproject">

	<allow pkg=".*" regex="true"/>

	<!-- Use BDD Mockito -->
	<disallow class="org.mockito.Mockito.when"/>
	<disallow class="org.mockito.Mockito.doThrow"/>
	<disallow class="org.mockito.Mockito.doAnswer"/>

...

```

**Comment by spencergibb** [comment]
Also use `com.google.common` package as that is guava's base package.

**Comment by spencergibb** [comment]
Thanks @philwebb! This will be super simple with you having done all of the hard work.

**Comment by spencergibb** [comment]
@philwebb where do you put `checkstyle.xml`?

**Comment by philwebb** [comment]
@spencergibb I have it in the same folder as `pom.xml`. You can probably use maven properties such as `baseDir` in `<configLocation>`.

**Comment by spencergibb** [comment]
hmm, ok.  Need to have it packaged up for use by other projects too.

**Comment by spencergibb** [comment]
@marcingrzejszczak I think you got this to work with zookeeper right (a shared checkstyle that is)?

**Comment by marcingrzejszczak** [comment]
@spencergibb our checkstyle is here https://github.com/spring-cloud/spring-cloud-build/blob/master/spring-cloud-build-tools/src/main/resources/checkstyle.xml . Here is the setup for Zookeeper - https://github.com/spring-cloud/spring-cloud-zookeeper/blob/master/pom.xml#L75-L101 . So it's enough to modify the checkstyle.xml from spring-cloud-build and all of the projects will be affected

**Comment by spencergibb** [comment]
Closed via 3d4369f6cb2219b2aaa41505540657ca6668c854
